pkg/internal/sqlsmith/type.go

```diff
@@ -11,19 +11,25 @@
 package sqlsmith
 
 import (
+	"context"
+
 	"github.com/cockroachdb/cockroach/pkg/sql/parser"
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/errors"
 )
 
-func typeFromName(name string) *types.T {
+func (s *Smither) typeFromName(name string) (*types.T, error) {
 	typRef, err := parser.ParseType(name)
 	if err != nil {
-		panic(errors.AssertionFailedf("failed to parse type: %v", name))
+		return nil, errors.AssertionFailedf("failed to parse type: %v", name)
+	}
+	typ, err := tree.ResolveType(context.TODO(), typRef, s)
```

Comment:
context.Background is fine. Uh or just nil. Am I a bad Go programmer?

Reply:
I'll stick with background :)